published/external/xdpapi_helper.h

```diff
@@ -0,0 +1,68 @@
+#pragma once
+
+#include "xdpapi.h"
+
+#define TEST_TIMEOUT_ASYNC_MS 1000
+#define POLL_INTERVAL_MS 10
+
+static
+VOID
+XdpDetach(
+    _In_ VOID *ClientContext
+    )
+{
+    UNREFERENCED_PARAMETER(ClientContext);
+}
+
+NTSTATUS
+InitializeXdpApi(
+    _Out_ XDP_API_CLIENT *ApiContext,
+    _Out_ const XDP_API_PROVIDER_DISPATCH **ProviderDispatch,
+    _Out_ const XDP_API_PROVIDER_BINDING_CONTEXT **ProviderBindingContext,
+    _In_ const XDP_API_CLIENT_DISPATCH *ClientDispatch,
+    _In_ UINT32 Version
+    )
+{
+    NTSTATUS Status;
+    KEVENT Event;
+    INT32 TimeoutMs = TEST_TIMEOUT_ASYNC_MS;
```

Comment:
If this is a public header, the timeout should be configurable. No timeout ("must not wait") and infinite timeout should both be supported.